### PR TITLE
Release notes and version string update.

### PR DESCRIPTION
**Pull request by gunan**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+
+## Breaking Changes
+* Prebuilt binaries are now built against CUDA 9 and cuDNN 7.
+* Our Linux binaries are built using ubuntu 16 containers, potentially
+  introducing glibc incompatibility issues with ubuntu 14.
+* Starting from 1.6 release, our prebuilt binaries will use AVX instructions.
+  This may break TF on older CPUs.
+
+## Major Features And Improvements
+* Eager execution preview version is available under `tensorflow/contrib/eager/`.
```
Comment (asimshankar) [inline RELEASE.md]: Should we make this a link?
[Eager execution](https://research.googleblog.com/2017/10/eager-execution-imperative-define-by.html)

And the same for TensorFlow lite, linking to https://www.tensorflow.org/mobile/tflite/?

**Review by asimshankar — APPROVED** [review]

File: tensorflow/core/public/version.h [hunk 2]
```diff
@@ -19,8 +19,8 @@ limitations under the License.
 // TensorFlow uses semantic versioning, see http://semver.org/.
 
 #define TF_MAJOR_VERSION 1
-#define TF_MINOR_VERSION 4
-#define TF_PATCH_VERSION 0
+#define TF_MINOR_VERSION 5
+#define TF_PATCH_VERSION 0rc0
```
Comment (asimshankar) [inline tensorflow/core/public/version.h]: Isn't `TF_PATCH_VERSION` supposed to be 0 and `-rc0` should go in the `TF_VERSION_SUFFIX`?

File: tensorflow/docs_src/install/install_c.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ enable TensorFlow for C:
          OS="linux" # Change to "darwin" for macOS
          TARGET_DIRECTORY="/usr/local"
          curl -L \
-           "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-${TF_TYPE}-${OS}-x86_64-1.4.0.tar.gz" |
+           "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-${TF_TYPE}-${OS}-x86_64-1.5.0rc0.tar.gz" |
```
Comment (asimshankar) [inline tensorflow/docs_src/install/install_c.md]: The filename should be `-rc0.tar.gz`

**Review by asimshankar — CHANGES_REQUESTED** [review]
Sorry, I was too eager with the approval earlier :)

I think the filenames and versions need to be fixed so that the version suffix is `-rc0`

**Comment by gunan** [comment]
Should be all addressed now. I messed up the version string update, but it should be good now.

**Review by av8ramit — APPROVED** [review]

File: tensorflow/docs_src/install/install_sources.md [hunk 4]
```diff
@@ -456,8 +456,8 @@ Stack Overflow and specify the `tensorflow` tag.
 **Linux**
 <table>
 <tr><th>Version:</th><th>CPU/GPU:</th><th>Python Version:</th><th>Compiler:</th><th>Build Tools:</th><th>cuDNN:</th><th>CUDA:</th></tr>
-<tr><td>tensorflow-1.4.0</td><td>CPU</td><td>2.7, 3.3-3.6</td><td>GCC 4.8</td><td>Bazel 0.5.4</td><td>N/A</td><td>N/A</td></tr>
-<tr><td>tensorflow_gpu-1.4.0</td><td>GPU</td><td>2.7, 3.3-3.6</td><td>GCC 4.8</td><td>Bazel 0.5.4</td><td>6</td><td>8</td></tr>
+<tr><td>tensorflow-1.5.0rc0</td><td>CPU</td><td>2.7, 3.3-3.6</td><td>GCC 4.8</td><td>Bazel 0.5.4</td><td>N/A</td><td>N/A</td></tr>
```
Comment (av8ramit) [inline tensorflow/docs_src/install/install_sources.md]: This is an unfortunate issue with the script that changes the reference to 1.4.0. A fix is on my todo but can you revert it here to 1.4.0 if you get a chance? I've also added a note to the release process.

**Review by asimshankar — DISMISSED** [review]

File: tensorflow/docs_src/install/install_sources.md (hunk 4 above)
Comment (av8ramit) [inline tensorflow/docs_src/install/install_sources.md]: This is an unfortunate issue with the script that changes the reference to 1.4.0. A fix is on my todo but can you revert it here to 1.4.0 if you get a chance? I've also added a note to the release process.

**Review by av8ramit — CHANGES_REQUESTED** [review]

File: tensorflow/docs_src/install/install_sources.md (hunk 4 above)
Comment (gunan) [reply tensorflow/docs_src/install/install_sources.md]: Done.

File: tensorflow/docs_src/install/install_sources.md [hunk 5]
```diff
@@ -483,8 +483,8 @@ Stack Overflow and specify the `tensorflow` tag.
 **Windows**
 <table>
 <tr><th>Version:</th><th>CPU/GPU:</th><th>Python Version:</th><th>Compiler:</th><th>Build Tools:</th><th>cuDNN:</th><th>CUDA:</th></tr>
-<tr><td>tensorflow-1.4.0</td><td>CPU</td><td>3.5-3.6</td><td>MSVC 2015 update 3</td><td>Cmake v3.6.3</td><td>N/A</td><td>N/A</td></tr>
-<tr><td>tensorflow_gpu-1.4.0</td><td>GPU</td><td>3.5-3.6</td><td>MSVC 2015 update 3</td><td>Cmake v3.6.3</td><td>6</td><td>8</td></tr>
+<tr><td>tensorflow-1.5.0rc0</td><td>CPU</td><td>3.5-3.6</td><td>MSVC 2015 update 3</td><td>Cmake v3.6.3</td><td>N/A</td><td>N/A</td></tr>
```
Comment (av8ramit) [inline tensorflow/docs_src/install/install_sources.md]: Here as well.

Comment (gunan) [reply tensorflow/docs_src/install/install_sources.md]: Done

File: tensorflow/docs_src/install/install_sources.md [hunk 6]
```diff
@@ -471,7 +471,7 @@ Stack Overflow and specify the `tensorflow` tag.
 **Mac**
 <table>
 <tr><th>Version:</th><th>CPU/GPU:</th><th>Python Version:</th><th>Compiler:</th><th>Build Tools:</th><th>cuDNN:</th><th>CUDA:</th></tr>
-<tr><td>tensorflow-1.4.0</td><td>CPU</td><td>2.7, 3.3-3.6</td><td>Clang from xcode</td><td>Bazel 0.5.4</td><td>N/A</td><td>N/A</td></tr>
+<tr><td>tensorflow-1.5.0rc0</td><td>CPU</td><td>2.7, 3.3-3.6</td><td>Clang from xcode</td><td>Bazel 0.5.4</td><td>N/A</td><td>N/A</td></tr>
```
[the remark this answers is not in the document]
Comment (gunan) [reply tensorflow/docs_src/install/install_sources.md]: Done

**Review by av8ramit — APPROVED** [review]

**Pull request merged**